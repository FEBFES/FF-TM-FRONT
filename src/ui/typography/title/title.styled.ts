import styled, { RuleSet, css } from 'styled-components';
import { TitleLevel } from './title';

export type StyledVariants = {
  [key in TitleLevel]?: RuleSet;
};

const sizes: StyledVariants = {
  h1: css`
    font-weight: 600;
    font-size: 38px;
    color: var(--font-gray);
  `,
  h2: css`
    font-weight: 500;
    font-size: 30px;
    color: var(--font-gray);
  `,
  h3: css`
    font-weight: 500;
    font-size: 24px;
    color: var(--font-gray);
  `,
  h4: css`
    font-weight: 500;
    font-size: 20px;
    color: var(--font-gray);
  `,
  h5: css`
    font-weight: 500;
    font-size: 16px;
    color: var(--font-gray);
  `,
  h6: css`
    font-weight: 500;
    font-size: 14px;
    color: var(--font-gray);
  `,
};

export const STitle = styled.h1<{
  size: TitleLevel;
  cursor?: 'pointer';
  hover?: 'underline';
}>`
  ${({ size }) => sizes[size]};
  cursor: ${({ cursor }) => cursor === 'pointer' && 'pointer'};

  &: hover {
    text-decoration: ${({ hover }) => hover === 'underline' && 'underline'};
  }
`;
