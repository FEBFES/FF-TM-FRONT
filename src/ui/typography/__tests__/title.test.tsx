import React from 'react';
import { Title } from '../title/title';
import renderer from 'react-test-renderer';

describe('<Title />', () => {
  it('SNAPSHOTS: UI - Title (h1) component - render', () => {
    const elem = renderer
      .create(
        <Title hover="underline" cursor="pointer">
          linka
        </Title>
      )
      .toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Title (h1) component - render', () => {
    const elem = renderer.create(<Title level={'h1'}>h1</Title>).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Title (h2) component - render', () => {
    const elem = renderer.create(<Title level={'h2'}>h2</Title>).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Title (h3) component - render', () => {
    const elem = renderer.create(<Title level={'h3'}>h3</Title>).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Title (h4) component - render', () => {
    const elem = renderer.create(<Title level={'h4'}>h4</Title>).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Title (h5) component - render', () => {
    const elem = renderer.create(<Title level={'h5'}>h5</Title>).toJSON();
    expect(elem).toMatchSnapshot();
  });

  it('SNAPSHOTS: UI - Title (h6) component - render', () => {
    const elem = renderer.create(<Title level={'h6'}>h6</Title>).toJSON();
    expect(elem).toMatchSnapshot();
  });
});
