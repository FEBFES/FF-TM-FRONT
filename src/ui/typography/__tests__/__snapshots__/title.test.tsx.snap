// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Title /> SNAPSHOTS: UI - Title (h1) component - render 1`] = `
<h1
  className="sc-bcPKhP dbDnGD"
  hover="underline"
  size="h3"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h1) component - render 2`] = `
<h1
  className="sc-bcPKhP djHSJu"
  cursor="pointer"
  size="h3"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h1) component - render 3`] = `
<h1
  className="sc-bcPKhP gtkEuv"
  size="h1"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h2) component - render 1`] = `
<h1
  className="sc-bcPKhP eJjWaU"
  size="h2"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h3) component - render 1`] = `
<h1
  className="sc-bcPKhP cFEvjN"
  size="h3"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h4) component - render 1`] = `
<h1
  className="sc-bcPKhP bwaqqp"
  size="h4"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h5) component - render 1`] = `
<h1
  className="sc-bcPKhP fapzAs"
  size="h5"
>
  linka
</h1>
`;

exports[`<Title /> SNAPSHOTS: UI - Title (h6) component - render 1`] = `
<h1
  className="sc-bcPKhP imMJyy"
  size="h6"
>
  linka
</h1>
`;
